components/motor/gpio/motor_encoder.go

```diff
@@ -102,15 +102,16 @@ func newEncodedMotor(
 	em.encoder = realEncoder
 
 	if len(motorConfig.ControlLoop.Blocks) != 0 {
-		cLoop, err := control.NewLoop(logger, motorConfig.ControlLoop, em)
-		if err != nil {
+		if err = em.FindControlBlock(); err != nil {
 			return nil, err
 		}
-		err = cLoop.Start()
+		pidBlock, err := em.loop.ConfigAt(cancelCtx, "PID")
```

Comment:
Have you observed the loop canceling the context if misconfigured in the config or given bad inputs such as `NaN` input speeds?

Reply:
I have not. when I didn't have the config set up correctly it just didn't work. I added PID to the validate function I am working on to make sure that's also created correctly